crates/mun_codegen/src/ir/body.rs

```diff
@@ -585,6 +586,61 @@ impl<'a, 'b, D: IrDatabase> BodyIrGenerator<'a, 'b, D> {
         }
     }
 
+    /// Generates IR to calculate a unary operation on an expression.
+    fn gen_unary_op(&mut self, expr: ExprId, op: UnaryOp) -> Option<BasicValueEnum> {
+        let ty = self.infer[expr].clone();
+        match ty.as_simple() {
+            Some(TypeCtor::Float(_ty)) => self.gen_unary_op_float(expr, op),
+            Some(TypeCtor::Int(ty)) => self.gen_unary_op_int(expr, op, ty.signedness),
```

Comment:
I think its important to ensure that the `Int` type is unsigned. You can match on the signedness of `ty`.

Reply:
Should it return `None`, or should it just use `unimplemented!`?

Reply:
Panic might be better, its invalid to ever get there.